test/e2e/main_test.go

```diff
@@ -85,8 +85,9 @@ func TestAllNS(t *testing.T) {
 	defer ctx.Cleanup(t)
 
 	ns := ctx.CreateNamespace(t, framework.KubeClient)
+	stableVersion := "quay.io/prometheus-operator/prometheus-operator:v0.47.1"
```

Comment:
It would be better to read [`VERSION`](https://github.com/prometheus-operator/prometheus-operator/blob/master/VERSION) file, use [github.com/blang/semver/v4](https://github.com/blang/semver) (we are already using it for semver comparisons), and ensure we are always using `X.(Y-1).0` version instead of static 0.47.1. Patch component can be either set to `0` or ideally, we could do a full detection of last version by:
1) taking `major` and `minor` components of version string from VERSION file
2) compose branch string out of them (schema is `release-MAJOR.MINOR`)
3) use VERSION file from the composed branch (either by using `git` or by fetching from github)
4) compose container name as `quay.io/prometheus-operator/prometheus-operator:v + VERSION`

Reply:
OK. I also need to rewrite an upgrade e2e test method, separate from other e2e processes.